deluge/core/core.py

```diff
@@ -163,6 +163,12 @@ def __init__(self, listen_interface=None, read_only_config_keys=None):
                 self.config['listen_interface'] = listen_interface
             else:
                 log.error('Invalid listen interface (must be IP Address): %s', listen_interface)
+        if outgoing_interface:
+            if deluge.common.is_ip(outgoing_interface):
+                self.__old_interface = self.config['outgoing_interface']
```

Comment:
This overrides the listen_interface, so it either needs to be renamed (and logic added below to restore the config value) or removed.